src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftBuilder.java

```diff
@@ -78,26 +80,34 @@ public String getCheckForTriggeredDeployments() {
 	}
 	
 	public String getWaitTime() {
-		return waitTime;
+		return waitTime + waitUnit;
 	}
 	
 	@DataBoundSetter public void setWaitTime(String waitTime) {
 		this.waitTime = waitTime;
 	}
+
+	public String getWaitUnit() {
+        return waitUnit;
+    }
+
+    @DataBoundSetter public void setWaitUnit(String waitUnit) {
+        this.waitUnit = waitUnit;
+    }
 	
 	public String getWaitTime(Map<String,String> overrides) {
 		String val = getOverride(getWaitTime(), overrides);
 		if (val.length() > 0)
 			return val;
-		return Long.toString(GlobalConfig.getBuildWait());
+		return GlobalConfig.getBuildWait() + GlobalConfig.getBuildWaitUnits();
 	}
-	
-	
+
     private static final Logger LOGGER = Logger.getLogger(OpenShiftBuilder.class.getName());
 
 
 	@Extension
     public static class DescriptorImpl extends AbstractStepDescriptorImpl {
```

Comment:
I believe you are reworking the DSL side of things now, so for now I'm assuming you will rework the DSL descriptor impl's similarly wrt interfaces with default methods, etc. ... I'll defer on further review of the DSL package until you tell me the DSL stuff should be looked at.